build_projects/dotnet-cli-build/BuildFPMToolPreReqs.cs

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// This task prepares the command line parameters for running a RPM build using FPM tool and also updates the copyright and changelog file tokens.
+    /// If parses various values from the config json by first reading it into a model and then builds the required string for parameters and passes it back.
+    /// 
+    /// </summary>
+    public class BuildFPMToolPreReqs : Task
+    {
+        [Required]
+        public string InputDir { get; set; }
+        [Required]
+        public string OutputDir { get; set; }
+        [Required]
+        public string PackageVersion { get; set; }
+        [Required]
+        public string ConfigJsonFile { get; set; }
+        [Output]
+        public string FPMParameters { get; set; }
+        public override bool Execute()
+        {
+            try
+            {
+                if (!File.Exists(ConfigJsonFile))
+                {
+                    throw new FileNotFoundException($"Expected file {ConfigJsonFile} was not found.");
+                }
+
+                // Open the Config Json and read the values into the model
+                TextReader projectFileReader = File.OpenText(ConfigJsonFile);
+                if (projectFileReader != null)
```

Comment:
This file is a copy from https://github.com/dotnet/core-setup/blob/53771d8bfc8b19afbd45e5dbdcf727049d4f9231/tools-local/tasks/BuildFPMToolPreReqs%20.cs

However I should invest more time in it, since we will maintain this file in our repo. Let me fix it